CodeStarterCamp_Week3/main.swift

```diff
@@ -8,26 +8,35 @@
 
 import Foundation
 
-struct Person {
-    var name: String = ""
-    var money: Int = 0
+class Person {
+    var name: String
+    var money: Int
     
-    func buySome(coffee: Coffee) {
-        print("\(coffee)주세요.")
+    init(name: String, money: Int) {
+        self.name = name
+        self.money = money
     }
 }
 
 struct CoffeeShop {
     var sales: Int = 0
```

Comment:
- Person 을 class로 바꾸지 않았을때 오류가 생겼습니다. 
<img width="797" alt="스크린샷 2022-03-04 오전 4 22 32" src="https://user-images.githubusercontent.com/75517368/156719970-f186d138-ee43-458b-84ad-db6d0075bd51.png">

그래서
Person 은 다른 객체의 함수에서 값을 수정하기위해 주소값을 전달하는 class 로 선언하였습니다.


CoffeeShop 은 따로 다른 객체에서 값을 받아 수정하는 경우가 없어서 struct를 사용하였습니다.
찾아봤는데 
기본적으로 둘다 사용 가능 할 때는 struct 를 사용하라 라고 해서 Person 은 class 로 CoffeeShop은 struct 로 선언 하였습니다.
> Using structures makes it easier to reason about a portion of your code without needing to consider the whole state of your app. Because structures are value types—unlike classes—local changes to a structure aren't visible to the rest of your app unless you intentionally communicate those changes as part of the flow of your app. As a result, you can look at a section of code and be more confident that changes to instances in that section will be made explicitly, rather than being made invisibly from a tangentially related function call.


https://developer.apple.com/documentation/swift/choosing_between_structures_and_classes

Reply:
@hcooch2ch3  
어.... 다시해봤는데 `class` 를 사용해도 오류가 나지않아요..  😢 

`class`로 선언하여 `func` 를 사용해도 값이 나오고
`struct`로 선언하여 `mutating func` 사용해도 똑같이 값이 나오면 


`struct` 를 사용하는것이 더 좋은 방법일까요??
만약 선택지가 둘중 하나라면 🌰 은 어느 것을 사용 했을지가 궁금합니다! 🙏 🙏